evo_researcher/functions/research.py

```diff
@@ -11,10 +11,9 @@
 def research(
     goal: str,
     openai_key: str,
-    tavily_key: str,
     use_summaries: bool,
     model: str = "gpt-4-1106-preview",
-    initial_subqueries_limit: int = 20,
+    initial_subqueries_limit: int = 4,
```

Comment:
Changed because I noticed that most of the time, it just uses the first 4.

Reply:
Hmm, can't see this. Can you link an example?

Reply:
Sure!

Maybe I was more unlucky before, but still, in 3 out 5 goals, 3 questions are the same after reranking. In two of them, one is the same, and I wouldn't say the others are better or worse. So we could save the generation of 2*16 questions. But I'm not opinioned about this, so I'm happy to revert this line if you think 20 is better.

```python
goals = [ 
    "The best way to cook a steak",
    "What is a black hole?",
    "How to make a chocolate cake",
    "how to make a good cup of coffee",
    "how to train a chihuahua",
]
for goal in goals:
    openai_key = "..."
    all_queries = [x.strip() for x in generate_subqueries(query=goal, limit=20, api_key=openai_key)]
    chosen_queries = [x.strip() for x in rerank_subqueries(queries=all_queries, goal=goal, api_key=openai_key)]
    print("Common queries across all:", len(set(all_queries) & set(chosen_queries)))
    print("Common queries in first 4:", len(set(all_queries[:4]) & set(chosen_queries[:4])))
    for x, y in zip(all_queries[:4], chosen_queries[:4]):
        print(f"{x} -> {y}")
    print("---")
```

Results:

```
Common queries across all: 21
Common queries in first 4: 1
The best way to cook a steak -> The best way to cook a steak
best steak cuts for cooking -> grilling vs pan-frying steak
optimal steak thickness for grilling -> cast iron skillet for steak
steak marinating techniques -> high heat vs low heat steak cooking
---
Common queries across all: 19
Common queries in first 4: 3
What is a black hole? -> black hole definition astronomy
black hole definition astronomy -> how do black holes form
how do black holes form -> types of black holes
types of black holes -> stellar mass black holes
---
Common queries across all: 20
Common queries in first 4: 3
How to make a chocolate cake -> chocolate cake recipe easy
chocolate cake recipe easy -> moist chocolate cake preparation steps
best chocolate cake ingredients -> best chocolate cake ingredients
moist chocolate cake preparation steps -> chocolate cake baking temperature and time
---
Common queries across all: 21
Common queries in first 4: 1
how to make a good cup of coffee -> how to make a good cup of coffee
best coffee bean varieties -> coffee brewing methods comparison
optimal coffee bean roast levels -> water quality for coffee brewing
coffee grinding techniques -> optimal water temperature for coffee extraction
---
Common queries across all: 21
Common queries in first 4: 4
how to train a chihuahua -> how to train a chihuahua
chihuahua training basics -> chihuahua training basics
chihuahua puppy training tips -> chihuahua puppy training tips
best treats for training chihuahuas -> best treats for training chihuahuas
```

Reply:
I reverted it to not block this PR because of it and we can change it later in another PR